File: Final Project/Medivault Spring/MediVault-Spring/MediVault/src/main/java/com/medivault/entites/Patient.java
package com.medivault.entites;

import java.util.List;

import com.medivault.enums.Gender;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToMany;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "patients", uniqueConstraints = {
		@UniqueConstraint(name = "uk_patient_user_id", columnNames = "user_id")
})
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor

public class Patient {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	
	@OneToOne(fetch = FetchType.LAZY, optional = false)
	@JoinColumn(name = "user_id", nullable = false)
	private User user;
	
	@Column(nullable = false, length = 150)
	private String name;
	
	@Column(nullable = false)
	private Integer age;
	
	@Enumerated(EnumType.STRING)
	@Column(nullable = false)
	private Gender gender;
	
	@Column(nullable = false, length = 15)
	private String phonenumber;
	
	@Column(length = 10)
	private String bloodGroup;
	
	@Column(columnDefinition = "TEXT")
	private String allergies;
	
	@Column(columnDefinition = "TEXT")
	private String familyHistory;
	
	@OneToMany(mappedBy = "patient", cascade = CascadeType.ALL, orphanRemoval = true)
	private List<Consultation> consultations;
	
	@OneToMany(mappedBy = "patient", cascade = CascadeType.ALL, orphanRemoval = true)
	private List<Appointment> appointments;
	
	@OneToMany(mappedBy = "patient", cascade = CascadeType.ALL, orphanRemoval = true)
	private List<PatientFile> patientFiles;
	
	

}

File: Final Project/MediVault_ DotNet/MediVault/MediVault.Backend/Models/Patient.cs
namespace MediVault.Backend.Models
{
    public class Patient
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public User? User { get; set; }
        public string Name { get; set; } = string.Empty;
        public int Age { get; set; }
        public string Gender { get; set; } = string.Empty;
        public string PhoneNumber { get; set; } = string.Empty;
        public string BloodGroup { get; set; } = string.Empty;
        public string Allergies { get; set; } = string.Empty;
        public string FamilyHistory { get; set; } = string.Empty;
        
        // Navigation property for consultations
        public ICollection<Consultation> Consultations { get; set; } = new List<Consultation>();
        
        // Navigation property for patient files
        public ICollection<PatientFile> PatientFiles { get; set; } = new List<PatientFile>();
        
        // Navigation property for appointments
        public ICollection<Appointment> Appointments { get; set; } = new List<Appointment>();
    }
}
